### ISSUE TITLE
[Enhancement]: Manage opensearch package association version with terraform

### ISSUE THREAD

**Issue by miguellobato84** [hashicorp/terraform-provider-aws#38381]
### Description

As of now, the aws_opensearch_package_association terraform resource doesn't have any reference to the package version, which means that there is no way to update the association by using terraform (you can do it via aws-cli or the UI).

By including a version attribute we can manage properly the updates from terraform

### Affected Resource(s) and/or Data Source(s)

aws_opensearch_package_association

### Potential Terraform Configuration

```terraform
resource "aws_opensearch_package" "example" {
  package_name = "example-txt"
  package_source {
    s3_bucket_name = aws_s3_bucket.my_opensearch_packages.bucket
    s3_key         = aws_s3_object.example.key
  }
  package_type = "TXT-DICTIONARY"
}

resource "aws_opensearch_package_association" "example" {
  package_id  = aws_opensearch_package.example.id
  domain_name = aws_opensearch_domain.my_domain.domain_name
  version = aws_opensearch_package.example.available_package_version
}
```

### Would you like to implement a fix?

No

**Comment by acwwat** [comment]
@miguellobato84 Could you please provide specific example where you can specify a version when associating a package in AWS Management Console and the AWS CLI?

Looking at [AssociatePackage](https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_AssociatePackage.html), it only supports the domain name and package ID as the arguments. Each package has a unique ID and represents a specific version (a.k.a. available package version), so the version is implied.

As for your configuration, I am guessing that if `aws_opensearch_package.example` is updated, it would have generated a new package ID and thus trigger an update for `aws_opensearch_package_association.example`. Is this not what you are observing, or do you have other requirements for a version (e.g. you want to associate an older version)?

**Comment by miguellobato84** [comment]
not exactly, from the UI you can update the package getting a new version while maintaining the same package id

**Comment by justinretzolk** [comment]
Related #39363
